compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -2162,9 +2168,23 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                             span: proper_span,
                             current: 0,
                             found: 0,
+                            prop_expr: None,
                         };
                         visitor.visit_stmt(stmt);
+
+                        let typeck_results = self.infcx.tcx.typeck(self.mir_def_id());
+                        let expr_ty: Option<Ty<'_>> = visitor.prop_expr.map(|expr| typeck_results.expr_ty(expr).peel_refs());
+
+                        let is_format_arguments_item =
+                            if let Some(expr_ty) = expr_ty
+                               && let ty::Adt(adt, _) = expr_ty.kind() {
+                                    self.infcx.tcx.lang_items().get(LangItem::FormatArguments) == Some(adt.did())
+                               } else {
+                                   false
+                               };
+
                         if visitor.found == 0
+                            && !is_format_arguments_item
```

Comment:
If we already special-case for `FormatArguments`, can't we also provide a more helpful diagnostic, which explains why this fails, instead of just getting rid of the binding suggestion? Note that we also still have this note: `= note: consider using a `let` binding to create a longer lived value`, so it might make sense to special case this.

Reply:
Good point!
I add a special note for `format_args!`, I'm not sure whether the words are accurate enough, seems it's hard to explain 😂 , so I added a link to the document.